igvm/commands.py

```diff
@@ -611,12 +611,20 @@ def _get_hypervisor(hostname, ignore_reserved=False):
 
 @contextmanager
 def _get_best_hypervisor(vm, hypervisor_states, offline=False):
-    hypervisors = (Hypervisor(o) for o in Query({
+    hypervisors = list()
+    for hv_obj in Query({
         'servertype': 'hypervisor',
         'environment': environ.get('IGVM_MODE', 'production'),
         'vlan_networks': vm.dataset_obj['route_network'],
         'state': Any(*hypervisor_states),
-    }, HYPERVISOR_ATTRIBUTES))
+    }, HYPERVISOR_ATTRIBUTES):
+        try:
+            hypervisors.append(Hypervisor(hv_obj))
+        except libvirtError:
```

Comment:
Catching of the libvirtError is not directly related to the commits you linked, they can be throws for lots of reasons and it's a good Idea to skip broken HVs instead of crashing igvm here since we usually have enough HVs.

Regarding those commits you mentioned: Before those we got all LVs via fabric run commands in get_logical_volumes in the Hypervisor constructor. Now we get them via our single ssh+libvirt connection per hypervisor. Since we cache and reuse those libvirt connections we are opening less connections overall now, no?

Reply:
If we handle exceptions only when given libvirt function is needed, we will have no way of fall back, at least with reasonable code. It is just safer to skip inaccessible HWLBs.